tools/plugin_checker.py

```diff
@@ -0,0 +1,188 @@
+                f.close()
+
+def check_plugins_for_updates():
+    '''Iterate through all plugins in PLUGIN_VERSION_DICT and check if any of
+    them need to be updated. If so, file an issue.'''
+    global PLUGINS_TO_UPDATE
+    global PLUGINS_FAILED_TO_CHECK
+    global PLUGINS_UP_TO_DATE
+    
+    for k, v in PLUGIN_VERSION_DICT.items():
+        if not args.silent:
+            print("Checking ", k)
+        if "github" not in v[1]:
+            PLUGINS_FAILED_TO_CHECK = PLUGINS_FAILED_TO_CHECK + k + "\n"
+            continue
+
+        # Some repo names have a trailing slash, which needs to be removed
+        split_url = v[1].split("/")
+        if (split_url[-1] == ""):
+            owner = split_url[-3]
+            repo = split_url[-2]
+        else:
+            owner = split_url[-2]
+            repo = split_url[-1]
+
+        # Some repo names have a trailing .git, which needs to be removed
+        if ".git" in repo:
+            repo = repo.split(".git")[0]
+
+        # Final URL to get repo data from
+        url = 'https://api.github.com/repos/%s/%s/releases/latest' % (owner, repo)
+        
+        # Create an authenticated session to fetch repo data
+        session = requests.Session()
+        headers = {
+            "Authorization": "token %s" % TOKEN,
+        }
+        r = session.get(url, headers=headers)
+
+        if (r.status_code == 200):
+            repo_data = r.json()
+            # Sometimes the version will be specified in the "name" or "tag_name" fields.
+            # We'll check for both just in case.
+            if compare_versions(v[0], repo_data['name']):
+                PLUGINS_TO_UPDATE = PLUGINS_TO_UPDATE + (k + " needs to be updated to " + repo_data['name']) + "\n"
+                title = k + " needs to be updated to " + repo_data['name']
+                body = "A new version of the " + k + " plugin is available upstream at:\n" + v[1]
+                if args.file:
+                    make_github_issue(title, body, REPO_LABELS)
+            elif compare_versions(v[0], repo_data['tag_name']):
+                PLUGINS_TO_UPDATE = PLUGINS_TO_UPDATE + (k + " needs to be updated to " + repo_data['tag_name']) + "\n"
+                title = k + " needs to be updated to " + repo_data['tag_name']
+                body = "A new version of the " + k + " plugin is available upstream at:\n" + v[1]
+                if args.file:
+                    make_github_issue(title, body, REPO_LABELS)
+            else:
+                PLUGINS_UP_TO_DATE = PLUGINS_UP_TO_DATE + k + "\n"
+        else:
+            PLUGINS_FAILED_TO_CHECK = PLUGINS_FAILED_TO_CHECK + k + "\n"
+    
+    # Update report messages
+    PLUGINS_TO_UPDATE = PLUGINS_TO_UPDATE + "\n"
+    PLUGINS_FAILED_TO_CHECK = PLUGINS_FAILED_TO_CHECK + "\n"
+    PLUGINS_UP_TO_DATE = PLUGINS_UP_TO_DATE + "\n"
+
+def compare_versions(base_version, compared_version):
+    '''Return True iff base_version is less than compared_version.'''   
```

Comment:
```suggestion
    '''Return True if base_version is less than compared_version.'''   
```

Reply:
This isn't a typo, it's meant to signify ["if and only if"](https://en.wikipedia.org/wiki/If_and_only_if#Origin_of_iff_and_pronunciation)

Reply:
ok might be better to fully write the word